web-src/src/components/App.js

```diff
@@ -67,18 +72,44 @@ function NoAccessMessage() {
 
 export function App() {
   const [hasConsent, setConsent] = React.useState(true);
+  const [mainSidePanelWidth, setMainSidePanelWidth] = React.useState(MAIN_SIDE_PANEL_EXPAND_WIDTH);
 
   const viewType = useRecoilValue(viewTypeState);
+  const [mainSidePanel, setMainSidePanelState] = useRecoilState(mainSidePanelState);
+
+  React.useEffect(() => {
+    const handleResize = () => {
+      setMainSidePanelState(window.innerWidth <= MAIN_SIDE_PANEL_COLLAPSE_WIDTH_THRESHOLD
+        ? MainSidePanelType.Collapsed
+        : MainSidePanelType.Expanded);
+      setMainSidePanelWidth(window.innerWidth <= MAIN_SIDE_PANEL_COLLAPSE_WIDTH_THRESHOLD
+        ? MAIN_SIDE_PANEL_COLLAPSE_WIDTH
+        : MAIN_SIDE_PANEL_EXPAND_WIDTH);
+    };
+    window.addEventListener('resize', handleResize);
```

Comment:
Any chance to use CSS responsive media queries or is making react that impossible? `@media (min-width: 1600px) {`

Reply:
The panel sizes are set by the `columns` prop using Grid component's capabilities:
`<Grid columns={[mainSidePanelWidth, '1fr']} ...`